core/src/storage2/collections/binary_heap/children.rs

```diff
@@ -96,16 +96,20 @@ where
         Self { left, right }
     }
 
-    /// Returns `true` if no child exists in this object, otherwise `false`.
-    pub fn is_empty(&self) -> bool {
-        self.left.is_none() && self.right.is_none()
+    /// Returns the number of existent children in this object.
+    pub fn count(&self) -> usize {
+        CHILDREN_PER_NODE as usize
+            - [self.left.as_ref(), self.right.as_ref()]
+                .iter()
+                .flatten()
+                .count()
     }
 
     /// Returns a shared reference to the element at `which`.
-    pub fn child(&self, which: ChildPosition) -> Option<&T> {
+    pub fn child(&self, which: ChildPosition) -> &Option<T> {
```

Comment:
Could you please briefly explain why you opted to use the less user friendly `&Option<T>` instead of the more user friendly `Option<&T>` here?

Reply:
As to make both API's `child()` and `child_mut()` consistently return a reference (`&Option<T>` vs. `&mut Option<T>`). `child_mut()` needs to return a mutable ref to an `Option`, so that we can use `take()` to get ownership of the Options value.

Reply:
> `child_mut()` needs to return a mutable ref to an `Option`, so that we can use `take()` to get ownership of the `Options` value.

Please add this as rational to the docs of `child_mut`.